src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.ILLink.targets

```diff
@@ -221,17 +231,45 @@ Copyright (c) .NET Foundation. All rights reserved.
     </PropertyGroup>
 
     <ItemGroup>
-      <!-- Treat any assemblies that already have customized TrimMode as trimmable. -->
+      <!-- Set IsTrimmable for any assemblies that already have customized TrimMode. -->
       <ManagedAssemblyToLink Condition=" '%(ManagedAssemblyToLink.TrimMode)' != '' ">
         <IsTrimmable>true</IsTrimmable>
       </ManagedAssemblyToLink>
-      <!-- Root and copy non-trimmable assemblies. -->
+
+      <!-- SetIsTrimmable for any assemblies listed in TrimmableAssembly. -->
+      <_ManagedAssemblyToLinkName Include="@(ManagedAssemblyToLink->'%(Filename)')">
+        <OriginalItemSpec>%(Identity)</OriginalItemSpec>
+      </_ManagedAssemblyToLinkName>
+      <_NonTrimmableManagedAssemblyToLinkName Include="@(_ManagedAssemblyToLinkName)" Exclude="@(TrimmableAssembly)" />
```

Comment:
Should this print a warning if the assembly is marked as trimmable, but it does not get trimmed?

Reply:
Hmm, I can see that being useful, but I'm not sure it should be a warning - since this is the same option that would be used to opt in to member-level trimming for an assembly.